docs/DeveloperGuide.md

```diff
@@ -132,13 +132,20 @@ and their interactions.
 
 #### How the components interact with each other
 The *Component Interaction Diagram* shows the inner workings of how each component in WerkIt interacts.
-The diagram depicts a scenario where a user attempts to create a workout, `workout /new sit up /reps 5`.
 
 ![Architecture Sequence Diagram](uml/sequenceDiagrams/miscellaneous/images/ArchitectureSequenceDiagram.png)
 
 <span class="box info">:memo: To improve the diagram's readability, 
 some methods or parameters have been omitted.</span>
 
+1. When `WerkIt` class is initialize, `UI` class is called to ask and get the user input.
+2. The `Parser` class parses the user input and identifies the command type (e.g. plan/schedule/workout/exercise). Based
+   on the command type, corresponding `Command` object is created.
+3. By executed the command, command action will be identified and appropriate actions will be performed.
+4. After the execution process is finished, appropriate message will be printed to show user that the command 
+   is executed successfully.
+5. At last, the change will be written to local file.
```

Comment:
Here are some suggestions for your consideration:

- (Point 2) Perhaps the word 'corresponding' at the end of Point 2 can be prefixed with the word 'the'?
- (Point 3) Perhaps the sentence can be rephrased to "When the `Command` object is executed, a sequence of actions will be performed. The actions performed are dependent on the type of action specified by the user."
- (Point 4) "After the execution process has finished, an appropriate message will be printed to show **the** user that the command is executed successfully."
- (Point 5) "~At last~ **Finally**, the change ..."

Note that the bolded points are not meant to be actually bolded. They're just to indicate words that I've added on to your existing sentence :)

Reply:
Thank you for your suggestion. I will rephrase these points.